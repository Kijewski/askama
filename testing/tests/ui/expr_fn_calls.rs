#![allow(unused_variables)]

use askama::Template;

pub fn static_fn(arg: &str) -> &str { arg }
pub fn static_fn2(arg: &str) -> Option<&str> { Some(arg) }


#[derive(Template)]
#[template(source = "{{ test_fn(arg = 5) }}", ext = "html")]
struct NamedArgsInRustExprMemberFn;

impl NamedArgsInRustExprMemberFn {
    pub fn test_fn(&self, arg: u64) -> &'static str {
        "This is a test"
    }
}


#[derive(Template)]
#[template(source = r#"{{ self::static_fn(arg = "test") }}"#, ext = "html")]
struct NamedArgsInRustExprStaticCall;


#[derive(Template)]
#[template(source = r#"{{ self::static_fn2("test").unwrap(arg = "test") }}"#, ext = "html")]
struct NamedArgsInRustExprStaticCall2;


#[derive(Template)]
#[template(source = r#"{% let test = self::static_fn(arg = "test") %}"#, ext = "html")]
struct NamedArgsInRustExprStaticCall3;


fn main() {}
