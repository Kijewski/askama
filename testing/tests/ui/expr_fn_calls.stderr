error: failed to parse template source
 --> <source attribute>:1:15
       "= 5) }}"
  --> tests/ui/expr_fn_calls.rs:10:21
   |
10 | #[template(source = "{{ test_fn(arg = 5) }}", ext = "html")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^

error: failed to parse template source
 --> <source attribute>:1:23
       "= \"test\") }}"
  --> tests/ui/expr_fn_calls.rs:21:21
   |
21 | #[template(source = r#"{{ self::static_fn(arg = "test") }}"#, ext = "html")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: failed to parse template source
 --> <source attribute>:1:39
       "= \"test\") }}"
  --> tests/ui/expr_fn_calls.rs:26:21
   |
26 | #[template(source = r#"{{ self::static_fn2("test").unwrap(arg = "test") }}"#, ext = "html")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: failed to parse template source
 --> <source attribute>:1:34
       "= \"test\") %}"
  --> tests/ui/expr_fn_calls.rs:31:21
   |
31 | #[template(source = r#"{% let test = self::static_fn(arg = "test") %}"#, ext = "html")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
